CI: linalg: several test failures for ILP64

### Describe your issue.

It seems that several tests within the scipy-openblas ILP64 job are failing.

https://github.com/scipy/scipy/actions/runs/20587107260/job/59125393507

### Reproducing Code Example

```python
spin test -s linalg
```

### Error message

```shell
=========================== short test summary info ============================
FAILED scipy/interpolate/tests/test_interpolate.py::TestPPoly::test_roots_croots - AssertionError: 
Not equal to tolerance rtol=5.96046e-08, atol=1e-10

Mismatched elements: 13 / 1690 (0.769%)
First 5 mismatches are at indices:
 [86]: (-1.9557099545833975e-09+0j) (ACTUAL), 0j (DESIRED)
 [216]: (-1.6589372357484703e-09-7.698380706744713e-10j) (ACTUAL), 0j (DESIRED)
 [346]: (-1.6589372357484703e-09+7.698380706744713e-10j) (ACTUAL), 0j (DESIRED)
 [476]: (-3.688637304422297e-10+4.572760141383417e-11j) (ACTUAL), 0j (DESIRED)
 [606]: (-3.688637304422297e-10-4.572760141383417e-11j) (ACTUAL), 0j (DESIRED)
Max absolute difference among violations: 1.95570995e-09
Max relative difference among violations: inf
 ACTUAL: array([-4.100714e-17-4.552999e-16j,  1.196376e-15-1.640193e-16j,
       -2.879513e-16+0.000000e+00j, ..., -2.217009e-15+5.990403e-16j,
       -3.460614e-15+2.437666e-15j, -1.574004e-15+6.972096e-16j],
      shape=(1690,))
 DESIRED: array([0.+0.j, 0.+0.j, 0.+0.j, ..., 0.+0.j, 0.+0.j, 0.+0.j], shape=(1690,))
FAILED scipy/linalg/tests/test_solvers.py::TestSolveContinuousAre::test_solve_continuous_are[13-case13] - ValueError: Reordering of (A, B) failed because the transformed matrix pair (A, B) would be too far from generalized Schur form; the problem is very ill-conditioned. (A, B) may have been partially reordered.
FAILED scipy/linalg/tests/test_solvers.py::TestSolveContinuousAre::test_solve_continuous_are[17-case17] - AssertionError: 
Arrays are not almost equal to 12 decimals

Mismatched elements: 2159 / 10000 (21.6%)
First 5 mismatches are at indices:
 [0, 6]: 2.3889356017960648e-12 (ACTUAL), 0.0 (DESIRED)
 [0, 7]: -1.6678184455184178e-12 (ACTUAL), 0.0 (DESIRED)
 [0, 39]: 1.7782759654529884e-12 (ACTUAL), 0.0 (DESIRED)
 [0, 40]: -6.793582933427461e-12 (ACTUAL), 0.0 (DESIRED)
 [0, 41]: 1.0731864631045143e-11 (ACTUAL), 0.0 (DESIRED)
Max absolute difference among violations: 8.71292863e-07
Max relative difference among violations: inf
 ACTUAL: array([[-8.548879067399e-14,  1.209968562140e-13, -4.585403778252e-14,
        ..., -1.654958393819e-07,  6.176387159945e-07,
        -8.712928625207e-07],...
 DESIRED: array([[0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],...
FAILED scipy/linalg/tests/test_solvers.py::TestSolveDiscreteAre::test_solve_discrete_are[17-case17] - AssertionError: 
Not equal to tolerance rtol=1e-07, atol=1.5e-06

Mismatched elements: 3 / 16 (18.8%)
Mismatch at indices:
 [0, 0]: -0.39336096854514213 (ACTUAL), 0.0 (DESIRED)
 [0, 1]: -0.006146264784218554 (ACTUAL), 0.0 (DESIRED)
 [1, 0]: -0.006146264784218554 (ACTUAL), 0.0 (DESIRED)
Max absolute difference among violations: 0.39336097
Max relative difference among violations: inf
 ACTUAL: array([[-3.933610e-01, -6.146265e-03,  3.261350e-13,  2.182601e-16],
       [-6.146265e-03,  4.373123e-09, -2.178450e-14,  1.478318e-15],
       [ 3.261350e-13, -2.178450e-14, -1.110223e-16,  1.668322e-16],
       [ 2.182601e-16,  1.478318e-15,  1.668322e-16,  3.330669e-16]])
 DESIRED: array([[0., 0., 0., 0.],
       [0., 0., 0., 0.],
       [0., 0., 0., 0.],
       [0., 0., 0., 0.]])
FAILED scipy/linalg/tests/test_solvers.py::TestSolveDiscreteAre::test_solve_discrete_are[18-case18] - ValueError: Reordering of (A, B) failed because the transformed matrix pair (A, B) would be too far from generalized Schur form; the problem is very ill-conditioned. (A, B) may have been partially reordered.
= 5 failed, 68509 passed, 7460 skipped, 203 xfailed, 22 xpassed in 457.09s (0:07:37) =
```

### SciPy/NumPy/Python version and system information

```shell
Python 3.12.12
BLAS: SciPy OpenBLAS ILP64
OS: Ubuntu 24.04
```

--- a/.github/workflows/linux_blas.yml
+++ b/.github/workflows/linux_blas.yml
@@ -57,7 +57,11 @@ jobs:
         sudo apt-get install -y gfortran
         pip install cython numpy pybind11 pythran pytest hypothesis pytest-xdist pooch
         pip install -r requirements/dev.txt
-        pip install git+https://github.com/numpy/meson.git@main-numpymeson
+        # numpy fork of meson with meson v1.8.2 - not public, but works for our
+        # purposes here, we need its BLAS ILP64 support until that is merged
+        # in Meson upstream. Note that a more recent commit on this fork didn't work,
+        # see gh-24251.
+        pip install git+https://github.com/numpy/meson.git@e72c717199fa18d34020c7c97f9de3f388c5e055
         pip install mkl mkl-devel
 
     - name: Build with defaults (LP64)
@@ -92,7 +96,7 @@ jobs:
         sudo apt-get install -y gfortran
         pip install cython numpy pybind11 pythran pytest hypothesis pytest-xdist pooch
         pip install -r requirements/dev.txt
-        pip install git+https://github.com/numpy/meson.git@main-numpymeson
+        pip install git+https://github.com/numpy/meson.git@e72c717199fa18d34020c7c97f9de3f388c5e055
         pip install mkl mkl-devel
 
     - name: Build with ILP64
@@ -126,7 +130,7 @@ jobs:
         sudo apt-get install -y gfortran
         pip install cython numpy pybind11 pythran pytest hypothesis pytest-xdist pooch
         pip install -r requirements/dev.txt
-        pip install git+https://github.com/numpy/meson.git@main-numpymeson
+        pip install git+https://github.com/numpy/meson.git@e72c717199fa18d34020c7c97f9de3f388c5e055
         pip install scipy-openblas32 scipy-openblas64
 
     - name: Write out scipy-openblas64.pc

--- a/.github/workflows/linux_blas.yml
+++ b/.github/workflows/linux_blas.yml
@@ -131,7 +131,10 @@ jobs:
         pip install cython numpy pybind11 pythran pytest hypothesis pytest-xdist pooch
         pip install -r requirements/dev.txt
         pip install git+https://github.com/numpy/meson.git@e72c717199fa18d34020c7c97f9de3f388c5e055
-        pip install scipy-openblas32 scipy-openblas64
+        # Yes, we install both of scipy-openblas32|64, we need both LP64 and
+        # ILP64 (this is obviously a hack, and not supportable outside of SciPy
+        # CI and manual dev env setup; see gh-21889 for details).
+        pip install -r requirements/openblas64.txt
 
     - name: Write out scipy-openblas64.pc
       run: |

--- a/requirements/openblas64.txt
+++ b/requirements/openblas64.txt
@@ -1,0 +1,2 @@
+scipy-openblas32==0.3.30.0.8
+scipy-openblas64==0.3.30.0.8

